snuba/web/split.py

```diff
@@ -57,40 +59,68 @@ def time_split(dataset, request: Request, *args, **kwargs):
         into smaller increments, and start with the last one, so that we can potentially
         avoid querying the entire range.
         """
+        limit = request.query.get_limit()
+        if limit is None or request.query.get_groupby():
+            return None
+
+        if request.query.get_offset() >= 1000:
+            return None
+
+        orderby = request.query.get_orderby()
+        if not orderby or orderby[0] != f"-{self.__timestamp_col}":
+            return None
+
+        conditions = request.query.get_conditions() or []
+        from_date_str = next(
+            (
+                condition[2]
+                for condition in conditions
+                if _identify_condition(condition, self.__timestamp_col, ">=")
+            ),
+            None,
+        )
+
+        to_date_str = next(
+            (
+                condition[2]
+                for condition in conditions
+                if _identify_condition(condition, self.__timestamp_col, "<")
+            ),
+            None,
+        )
+
+        if not from_date_str or not to_date_str:
+            return None
+
         date_align, split_step = state.get_configs(
             [("date_align_seconds", 1), ("split_step", 3600)]  # default 1 hour
         )
+        to_date = util.parse_datetime(to_date_str, date_align)
+        from_date = util.parse_datetime(from_date_str, date_align)
 
-        query_limit = request.query.get_limit()
-        limit = query_limit if query_limit is not None else 0
         remaining_offset = request.query.get_offset()
 
-        to_date = util.parse_datetime(
-            request.extensions["timeseries"]["to_date"], date_align
-        )
-        from_date = util.parse_datetime(
-            request.extensions["timeseries"]["from_date"], date_align
-        )
-
         overall_result = None
         split_end = to_date
         split_start = max(split_end - timedelta(seconds=split_step), from_date)
         total_results = 0
         while split_start < split_end and total_results < limit:
-            request.extensions["timeseries"]["from_date"] = split_start.isoformat()
-            request.extensions["timeseries"]["to_date"] = split_end.isoformat()
+            # We need to make a copy to use during the query execution because we replace
+            # the start-end conditions on the query at each iteration of this loop.
+            split_request = copy.deepcopy(request)
```

Comment:
Processors may also mutate this too, right? The scope of mutation isn't just limited this function like this comment change implies.

Reply:
No, we are past the processors at this point. The only changes to the query are those done by this method, thus the comment.

Reply:
Sorry, my mistake, somehow I managed to forget that was the entire point of this change to begin with.